Avoid Unnecessary Tranpose?

edit to sparse/_common.py to improve tensordot speed for _dot_ndarray_coo
remove line 492: 
b = b.T

change lines 1147 and 1148 to:
oidx2 = coords2[1, didx2]
out[oidx1, oidx2] += array1[oidx1, coords[0, didx2]] * data2[didx2]

This avoids the creation of a new COO matrix during the transpose process, which can be time consuming. 
Is this messing up some normal usage of _dot_ndarray_coo? So far I have had no issues. 
Possibly this change could be added to other _dot* functions to avoid creating extraneous COO matrices?

Sorry, I don't know how to fork stuff quite yet, but I'll read up and try to fork if no one can think of a major issue with this change.

Following up:
I ended up only removing the transpose for the _dot_ndarray_coo_type function 
Using the tensordot benchmark for 100 runs:
New:  0.09302s +- 0.001313s
Old: 0.186462s +- 0.004102s

When I run the testing benchmark I only get back one more additional error than with a clean build, and it says
FAILED sparse/_common.py::BLACK
I am a little confused as to what means, but I am assuming it isn't dire because the clean build has 2 of the same type of error in it.

--- a/docs/conduct.rst
+++ b/docs/conduct.rst
@@ -137,7 +137,7 @@ available at
 https://www.contributor-covenant.org/version/2/0/code\_of\_conduct.html.
 
 Community Impact Guidelines were inspired by `Mozilla's code of conduct
-enforcement ladder <https://github.com/mozilla/diversity>`__.
+enforcement ladder <:ghuser:`mozilla/diversity`>`__.
 
 For answers to common questions about this code of conduct, see the FAQ
 at https://www.contributor-covenant.org/faq. Translations are available

--- a/docs/contributing.rst
+++ b/docs/contributing.rst
@@ -5,7 +5,7 @@ General Guidelines
 ------------------
 
 sparse is a community-driven project on GitHub. You can find our
-`repository on GitHub <https://github.com/pydata/sparse>`_. Feel
+`repository on GitHub <:ghuser:`pydata/sparse`>`_. Feel
 free to open issues for new features or bugs, or open a pull request
 to fix a bug or add a new feature.
 
@@ -22,7 +22,7 @@ Filing Issues
 -------------
 
 If you find a bug or would like a new feature, you might want to `consider
-filing a new issue on GitHub <https://github.com/pydata/sparse/issues>`_. Before
+filing a new issue on GitHub <:ghuser:`pydata/sparse/issues`>`_. Before
 you open a new issue, please make sure of the following:
 
 * This should go without saying, but make sure what you are requesting is within

--- a/sparse/_common.py
+++ b/sparse/_common.py
@@ -490,13 +490,13 @@ def _dot(a, b, return_type=None):
         return out
 
     if isinstance(a, np.ndarray) and isinstance(b, COO):
-        b = b.T
         a = a.view(type=np.ndarray)
 
         if return_type is None or return_type == np.ndarray:
             return _dot_ndarray_coo_type(a.dtype, b.dtype)(
                 a, b.coords, b.data, out_shape
             )
+        b = b.T
         coords, data = _dot_ndarray_coo_type_sparse(a.dtype, b.dtype)(
             a, b.coords, b.data, out_shape
         )
@@ -1148,8 +1148,8 @@ def _dot_ndarray_coo_type(dt1, dt2):
 
         for oidx1 in range(out_shape[0]):
             for didx2 in range(len(data2)):
-                oidx2 = coords2[0, didx2]
-                out[oidx1, oidx2] += array1[oidx1, coords2[1, didx2]] * data2[didx2]
+                oidx2 = coords2[1, didx2]
+                out[oidx1, oidx2] += array1[oidx1, coords2[0, didx2]] * data2[didx2]
 
         return out
 

--- a/sparse/_utils.py
+++ b/sparse/_utils.py
@@ -301,18 +301,18 @@ def equivalent(x, y):
 # copied from zarr
 # See https://github.com/zarr-developers/zarr-python/blob/master/zarr/util.py
 def human_readable_size(size):
-    if size < 2 ** 10:
+    if size < 2**10:
         return "%s" % size
-    elif size < 2 ** 20:
-        return "%.1fK" % (size / float(2 ** 10))
-    elif size < 2 ** 30:
-        return "%.1fM" % (size / float(2 ** 20))
-    elif size < 2 ** 40:
-        return "%.1fG" % (size / float(2 ** 30))
-    elif size < 2 ** 50:
-        return "%.1fT" % (size / float(2 ** 40))
+    elif size < 2**20:
+        return "%.1fK" % (size / float(2**10))
+    elif size < 2**30:
+        return "%.1fM" % (size / float(2**20))
+    elif size < 2**40:
+        return "%.1fG" % (size / float(2**30))
+    elif size < 2**50:
+        return "%.1fT" % (size / float(2**40))
     else:
-        return "%.1fP" % (size / float(2 ** 50))
+        return "%.1fP" % (size / float(2**50))
 
 
 def html_table(arr):

--- a/sparse/tests/test_coo.py
+++ b/sparse/tests/test_coo.py
@@ -694,10 +694,10 @@ def test_scalar_exponentiation():
     a = sparse.random((2, 3, 4), density=0.5)
     x = a.todense()
 
-    assert_eq(x ** 2, a ** 2)
-    assert_eq(x ** 0.5, a ** 0.5)
+    assert_eq(x**2, a**2)
+    assert_eq(x**0.5, a**0.5)
 
-    assert_eq(x ** -1, a ** -1)
+    assert_eq(x**-1, a**-1)
 
 
 def test_create_with_lists_of_tuples():

